mm2src/coins/nft/trading_proto_v2/mod.rs

```diff
@@ -0,0 +1,565 @@
+        if let Ok(decoded_params) = ethabi::decode(htlc_params(), data_bytes) {
+            if decoded_params[0] != Token::FixedBytes(expected_params.swap_id.clone()) {
+                return MmError::err(HtlcParamsError::WrongPaymentTx(format!(
+                    "Invalid 'swap_id' {:?}, expected {:?}",
+                    decoded_params[0],
+                    Token::FixedBytes(expected_params.swap_id)
+                )));
+            }
+            if decoded_params[1] != Token::Address(expected_params.taker_address) {
+                return MmError::err(HtlcParamsError::WrongPaymentTx(format!(
+                    "Invalid `taker_address` {:?}, expected {:?}",
+                    decoded_params[1],
+                    Token::Address(expected_params.taker_address)
+                )));
+            }
+            if decoded_params[2] != Token::Address(expected_params.token_address) {
+                return MmError::err(HtlcParamsError::WrongPaymentTx(format!(
+                    "Invalid `token_address` {:?}, expected {:?}",
+                    decoded_params[2],
+                    Token::Address(expected_params.token_address)
+                )));
+            }
+            if decoded_params[3] != Token::FixedBytes(expected_params.taker_secret_hash.clone()) {
+                return MmError::err(HtlcParamsError::WrongPaymentTx(format!(
+                    "Invalid 'taker_secret_hash' {:?}, expected {:?}",
+                    decoded_params[3],
+                    Token::FixedBytes(expected_params.taker_secret_hash)
+                )));
+            }
+            if decoded_params[4] != Token::FixedBytes(expected_params.maker_secret_hash.clone()) {
+                return MmError::err(HtlcParamsError::WrongPaymentTx(format!(
+                    "Invalid 'maker_secret_hash' {:?}, expected {:?}",
+                    decoded_params[4],
+                    Token::FixedBytes(expected_params.maker_secret_hash)
+                )));
+            }
+            if decoded_params[5] != Token::Uint(expected_params.time_lock) {
+                return MmError::err(HtlcParamsError::WrongPaymentTx(format!(
+                    "Invalid 'time_lock' {:?}, expected {:?}",
+                    decoded_params[5],
+                    Token::Uint(expected_params.time_lock)
+                )));
+            }
+        } else {
+            return MmError::err(HtlcParamsError::TxDeserializationError(
+                "Failed to decode HTLCParams from data_bytes".to_string(),
+            ));
+        }
+    } else {
+        return MmError::err(HtlcParamsError::TxDeserializationError(
+            "Expected Bytes for HTLCParams data".to_string(),
+        ));
+    }
+    Ok(())
+}
+
+// Representation of the Solidity HTLCParams struct.
+//
+// struct HTLCParams {
+//     bytes32 id;
+//     address taker;
+//     address tokenAddress;
+//     bytes32 takerSecretHash;
+//     bytes32 makerSecretHash;
+//     uint32 paymentLockTime;
+// }
```

Comment:
I think you meant for this to be a doc comment.

Reply:
done https://github.com/KomodoPlatform/komodo-defi-framework/pull/2084/commits/23d1c994e15eab82cd47fdf880a7ef81e7c6091c